### PR TITLE
[gcp] Add new `vault_gcp_secret_backend` resource for managing GCP

### PR DESCRIPTION
**Pull request by fatmcgav**
Secret backends [1]

Add docs.

[1] https://www.vaultproject.io/docs/secrets/gcp/index.html

### CODE REVIEW

**Comment by fatmcgav** [comment]
With this `Vault` patch applied locally:
```diff
diff --git a/vendor/github.com/hashicorp/vault/api/sys_mounts.go b/vendor/github.com/hashicorp/vault/api/sys_mounts.go
index 8ac5b45..740cbaf 100644
--- a/vendor/github.com/hashicorp/vault/api/sys_mounts.go
+++ b/vendor/github.com/hashicorp/vault/api/sys_mounts.go
@@ -135,6 +135,7 @@ type MountConfigInput struct {
        AuditNonHMACResponseKeys  []string          `json:"audit_non_hmac_response_keys,omitempty" mapstructure:"audit_non_hmac_response_keys"`
        ListingVisibility         string            `json:"listing_visibility,omitempty" mapstructure:"listing_visibility"`
        PassthroughRequestHeaders []string          `json:"passthrough_request_headers,omitempty" mapstructure:"passthrough_request_headers"`
+       Credentials               string            `json:"credentials,omitempty" mapstructure:"credentials"`
 }

 type MountOutput struct {
```

The tests pass as follows:
```
make testacc TEST=./vault TESTARGS="-run=TestGCPSecretBackend"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./vault -v -run=TestGCPSecretBackend -timeout 120m
=== RUN   TestGCPSecretBackend
--- PASS: TestGCPSecretBackend (0.06s)
PASS
ok      github.com/terraform-providers/terraform-provider-vault/vault   0.087s
```

File: vault/resource_gcp_secret_backend.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+	err := client.Sys().Mount(path, &api.MountInput{
+		Type:        "gcp",
+		Description: description,
+		Config: api.MountConfigInput{
+			DefaultLeaseTTL: fmt.Sprintf("%ds", defaultTTL),
+			MaxLeaseTTL:     fmt.Sprintf("%ds", maxTTL),
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("Error mounting to %q: %s", path, err)
```
Comment (tyrannosaurus-becks) [inline vault/resource_gcp_secret_backend.go]: Go prefers lower-cased error strings.

File: vault/resource_gcp_secret_backend.go [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+	d.SetPartial("description")
+	d.SetPartial("default_lease_ttl_seconds")
+	d.SetPartial("max_lease_ttl_seconds")
+
+	log.Printf("[DEBUG] Writing GCP configuration to %q", configPath)
+	data := map[string]interface{}{
+		"credentials": credentials,
+	}
+	if _, err := client.Logical().Write(configPath, data); err != nil {
+		return fmt.Errorf("Error writing GCP configuration for %q: %s", path, err)
```
Comment (tyrannosaurus-becks) [inline vault/resource_gcp_secret_backend.go]: Lower case please. :-)

File: vault/resource_gcp_secret_backend.go [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+	return gcpSecretBackendRead(d, meta)
+}
+
+func gcpSecretBackendRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	path := d.Id()
+
+	// TODO: Fill this out
```
Comment (tyrannosaurus-becks) [inline vault/resource_gcp_secret_backend.go]: Should this still be here?

File: vault/resource_gcp_secret_backend.go [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+	return nil
+}
+
+func gcpSecretBackendExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+	client := meta.(*api.Client)
+	path := d.Id()
+	log.Printf("[DEBUG] Checking if GCP backend exists at %q", path)
+	mounts, err := client.Sys().ListMounts()
+	if err != nil {
+		return true, fmt.Errorf("Error retrieving list of mounts: %s", err)
```
Comment (tyrannosaurus-becks) [inline vault/resource_gcp_secret_backend.go]: Lower case.

**Comment by fatmcgav** [comment]
Cheers for the review @tyrannosaurus-becks - All addressed :)

**Pull request merged**